traffic_portal/app/src/common/modules/form/job/form.job.tpl.html

```diff
@@ -47,14 +47,22 @@
                     <span ng-show="hasError(jobForm.regex)" class="form-control-feedback"><i class="fa fa-times"></i></span>
                 </div>
             </div>
-            <div class="form-group" ng-class="{'has-error': hasError(jobForm.ttl), 'has-feedback': hasError(jobForm.ttl)}">
+            <div class="form-group" ng-class="{'has-error': hasError(jobForm.ttlhours), 'has-feedback': hasError(jobForm.ttlhours)}">
                 <label class="control-label col-md-2 col-sm-2 col-xs-12">TTL (hours) *</label>
                 <div class="col-md-10 col-sm-10 col-xs-12">
-                    <input name="ttl" type="number" class="form-control" placeholder="Number of hours until the invalidation request expires" ng-model="job.ttl" ng-maxlength="3" ng-pattern="/^\d+$/" required autofocus>
-                    <small class="input-error" ng-show="hasPropertyError(jobForm.ttl, 'required')">Required Whole Number</small>
-                    <small class="input-error" ng-show="hasPropertyError(jobForm.ttl, 'maxlength')">Too Long</small>
-                    <small class="input-error" ng-show="hasPropertyError(jobForm.ttl, 'pattern')">Whole Number</small>
-                    <span ng-show="hasError(jobForm.ttl)" class="form-control-feedback"><i class="fa fa-times"></i></span>
+                    <input name="ttlhours" type="number" class="form-control" placeholder="Number of hours until the invalidation request expires" ng-model="job.ttlhours" ng-maxlength="3" ng-pattern="/^\d+$/" required autofocus>
```

Comment:
I'm just looking through this for docs reasons, but I can't stop myself from pointing out: this is an `input[type="number"]:required` so setting a pattern of `/^\d+$/` has no effect, since only things that would match that pattern (which can be just `pattern` too, since it's static - just like how `ng-maxlength=3` is no different from `maxlength=3` since `3` is a constant value but **also** also on a numeric input the controls will behave more nicely if you just set `max="999"` instead of `maxlength="3"`) are valid inputs anyway. And also, since this is a number, you don't need to be `parseInt`-ing down below.

Reply:
This is good stuff. I didn't know this. I honestly tried to touch as little as possible in this spot of the code, so what we see is the original markup. From what I am understanding, since it is `required` and `type="number"` we can do the following:
- Remove the `ng-pattern` attribute
- Replace the `ng-maxlength` with `max="999"` (Do/should we set a `min="0"` or `min="1"` as well?)

As for the `parseInt()` call in the controller, I was to understand that was parsing the input as it was returned from the server, which until this time was the `parameters` field and a string ("TTL:36h"). Is it safe to assume that since the API has been changed to return an uint it is now a JavaScript number and doesn't need to be "parsed"?

Reply:
Yes, the marshaler in Go will convert a `uint` to a JSON number, which `JSON.parse` will convert into a `number`.

There should probably be a minimum, probably 1.